orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -368,10 +317,10 @@ def forward(self, *inputs, **kwargs):
 
         # Exporting module to ONNX for the first time
         if not self._onnx_training:
-            if not self._device:
-                self._device = _utils.get_device_from_input_args_kwargs(self._original_module, *inputs, **kwargs)
-                if not self._device:
-                    raise RuntimeError('A device must be specified in the model or data!')
+            device_from_module = _utils.get_device_from_module(self._original_module)
+            if not self._device or self._device != device_from_module:
+                self._device = device_from_module
+                assert self._device is not None, RuntimeError('A device must be specified in the model or data!')
```

Comment:
This will now raise an assertion error as opposed to a runtime error. Is that the intention?

Reply:
Either one would be good, but I missed a`RuntimeError` in this line and it needs to be fixed :)